skglm/datafits/single_task.py

```diff
@@ -467,3 +467,52 @@ def gradient_scalar_sparse(self, X_data, X_indptr, X_indices, y, Xw, j):
 
     def intercept_update_self(self, y, Xw):
         pass
+
+
+class Gamma(BaseDatafit):
+    r"""Gamma datafit.
+
+    The datafit reads::
+
+    (1 / n_samples) * \sum_i (Xw_i + y_i * exp(-Xw_i))
```

Comment:
For the maths, I looked up GammaRegressor in ``scikit-learn`` code base,
(cf. [GammaRegressor](https://github.com/scikit-learn/scikit-learn/blob/f3f51f9b611bf873bd5836748647221480071a87/sklearn/linear_model/_glm/glm.py#L679), and [HalfGammaLoss](https://github.com/scikit-learn/scikit-learn/blob/9c4f023528ca5be585c0900b224f917a7435db74/sklearn/_loss/loss.py#L671-L689))
```shell
raw_prediction_i = Xw_i
loss(x_i) = log(exp(raw_prediction_i)/y_true_i) + y_true/exp(raw_prediction_i) - 1
```
Hence, by removing the constant, the formula of ``Gamma`` datafit is correct.

Reply:
@PABannier @Badr-MOUFAD do you think we should have the constant in the loss value ? It doesn't cost much and it harmonizes with sklearn

Reply:
I agree @mathurinm.
It intervenes in ``.value`` and we get rid of it in the formulas of grad and Hess.

Reply:
We modified the ``.value`` but forgot about updating that in the class docs